module/zfs/zfs_vnops.c

```diff
@@ -1429,6 +1429,7 @@ zfs_create(struct inode *dip, char *name, vattr_t *vap, int excl,
 			dmu_tx_hold_write(tx, DMU_NEW_OBJECT,
 			    0, acl_ids.z_aclp->z_acl_bytes);
 		}
+		dmu_tx_hold_zap(tx, zfsvfs->z_unlinkedobj, FALSE, NULL);
```

Comment:
This shouldn't be needed anymore after switching to `zfs_znode_delete`.  Here and elsewhere in this patch.

Reply:
Makes sense. Will take care of it.